### ISSUE TITLE
linter: False positive `typescript/consistent-return` when one branch is a `never`-returning function

### ISSUE THREAD

**Issue by justingrant** [oxc-project/oxc#24929]
### What version of Oxlint are you using?

1.73.0

### What command did you run?

`oxlint <file>` with `"typeAware": true` and the `typescript` plugin enabled.

### What does your `.oxlintrc.json` (or `oxlint.config.ts`) config file look like?

```jsonc
{
  "plugins": ["typescript"],
  "options": { "typeAware": true },
  "rules": {
    "typescript/consistent-return": "error"
  }
}
```

### What happened?

oxlint's `typescript/consistent-return` reports "expected a return value" for a function where one branch `return`s a value and the other branch ends in a call whose return type is `never` (e.g. `process.exit()`).

The upstream `@typescript-eslint/consistent-return` rule specifically extends the base ESLint rule to **add support for functions/branches that return `never`**, and does not flag this case. oxlint's port appears to be missing that exemption.

This reproduces even in a `.ts` file where type-aware analysis is active, so it's not a missing-type-info problem — the `never` type is available and the rule flags anyway.

Docs for the upstream behavior: https://typescript-eslint.io/rules/consistent-return/

```ts
export function readPackageLock(): unknown {
  try {
    return JSON.parse("{}");
  } catch (error) {
    console.error(error);
    process.exit(1); // returns `never`
  }
}
```

### Actual output
```
typescript(consistent-return): Function 'readPackageLock' expected a return value.
```

### Expected output

No diagnostic. The `catch` branch terminates in a `never`-returning call, so the function only ever returns a value or does not return at all — consistent with `@typescript-eslint/consistent-return`, which exempts `never`.

**Comment by camc314** [comment]
> The upstream @typescript-eslint/consistent-return rule specifically extends the base ESLint rule to add support for functions/branches that return never, and does not flag this case. oxlint's port appears to be missing that exemption.

can you expand/clarify? typescript-eslint reports an error on this code here: https://typescript-eslint.io/play/#ts=6.0.2&showAST=scope&fileType=.tsx&code=CYUwxgNghgTiAEAzArgOzAFwJYHtXwAcYcwQBnMgUQA8sMAKMHUAfgC55VkBbAIxBgBKDqhAA3AQG4AULJTpseeHCjAAClDABrKAHMQAGRJb6w%2BGi2ocAd3wBvafHgYYAT3gOnTuBmQx8AFIAygDyAHIAdASwZCD0AER2AL7xgjJOSfBgUBhgABbw9ALEQh6OXkyoZDgQIBHFODBFMCVp5U5EJORUtAwAjG0Z0klAA&eslintrc=N4KABGBEBOCuA2BTAzpAXGUEKQAIBcBPABxQGNoBLY-AWhXkoDt8B6MgeyeUuX0Ra1oifLGhN0URNGgdokcGAC%2BIJUA&tsconfig=N4KABGBEDGD2C2AHAlgGwKYCcDyiAuysAdgM6QBcYoEEkJemy0eAcgK6qoDCAFutAGsylBm3TgwAXxCSgA&tokens=false

**Comment by justingrant** [comment]
Hmm, weird. I must have confused this rule for a different rule that special-cases `never`. Sorry for missing this!

Regardless, correctly handling `never` would be useful!

**Comment by camc314** [comment]
No worries! That said, it's probably just better to use `noImplicitReturns` from typescript? http://oxc.rs/docs/guide/usage/linter/rules/typescript/consistent-return.html#typescript-consistent-return-oxlint

I'm thinking about closing this as not planned as using the type checker should do the same thing?

**Comment by justingrant** [comment]
Then maybe this rule should be deprecated?

**Comment by camc314** [comment]
> Then maybe this rule should be deprecated?

I don't know the full history here - we ported it to align with typescrip-eslint.

Closing as not planned due to the above.
